packages/shared/lib/db/migrations/20240403100610_add_account_status_info.cjs

```diff
@@ -0,0 +1,13 @@
+const tableName = '_nango_accounts';
+
+exports.up = async function (knex, _) {
+    await knex.schema.alterTable(tableName, function (table) {
+        table.boolean('is_capped').defaultTo(true);
```

Comment:
Open question:
I don't think it's really future proof, I'm not even sure what does that means in practice. Bastien will manually set this flag for people abusing the system or not paying?

imo future proof does not require much more work: a json column `plan` with something like 
`{ connection: { max: <number> }, syncs: { max: number }, records: { maxSize: number, maxTotalSize: number} }`
That would accommodate more use cases from: free tier, trial, different plan, custom entreprise, etc. Especially that we already know that it will happen for records storage and we'll need a place to store those info.

Reply:
I also vote for a `plan: json` column. Even if it means an object with a single boolean field for now. I can guarantee than very soon we will need different values and not only boolean but threshold values like @bodinsamuel described. Changing the type of the column will be painful then

Reply:
`is_capped` is simply if a check if a user is paying or not and also allows us to allow high priority trial customers get the full feature set. It is purposely simple and flexible. 

I disagree regarding the `json` column. I think making it a json column is anticipating something in which we have no idea what we're going to want and making for a check that is not as optimal as it can be given known parameters/spec in the meantime.

Reply:
it was an open suggestion, I don't think my solution was perfect but I do believe more planning would not hurt

Reply:
The planning for this was done as part of the spec of the [issue](https://linear.app/nango/issue/NAN-680/implement-way-to-upgradedowngrade-customers) & [project brief](https://linear.app/nango/document/resource-capping-v0-project-brief-749be67127a9) itself.